py/kubeflow/kubeflow/ci/kfctl_go_test_utils.py

```diff
@@ -0,0 +1,238 @@
+  """
+  url_for_spec = urllib.parse.urlparse(config_path)
+
+  if url_for_spec.scheme in ["http", "https"]:
+    data = requests.get(config_path)
+    return yaml.load(data.content)
+  else:
+    with open(config_path, 'r') as f:
+      config_spec = yaml.load(f)
+      return config_spec
+
+def set_env_init_args(use_basic_auth, use_istio):
+  # Is it really needed?
+  init_args = []
+  # Set ENV for basic auth username/password.
+  if use_basic_auth:
+    # Don't log the password.
+    # logging.info("Setting environment variables KUBEFLOW_USERNAME and KUBEFLOW_PASSWORD")
+    os.environ["KUBEFLOW_USERNAME"] = "kf-test-user"
+    os.environ["KUBEFLOW_PASSWORD"] = str(uuid.uuid4().hex)
+    init_args = ["--use_basic_auth"]
+  else:
+    # Owned by project kubeflow-ci-deployment.
+    logging.info("Setting environment variables CLIENT_SECRET and CLIENT_ID")
+    os.environ["CLIENT_SECRET"] = "CJ4qVPLTi0j0GJMkONj7Quwt"
+    os.environ["CLIENT_ID"] = (
+      "29647740582-7meo6c7a9a76jvg54j0g2lv8lrsb4l8g"
+      ".apps.googleusercontent.com")
+
+  if use_istio:
+    init_args.append("--use_istio")
+  else:
+    init_args.append("--use_istio=false")
+
+def filter_spartakus(spec):
+  """Filter our Spartakus from KfDef spec.
+
+  Args:
+    spec: KfDef spec
+
+  Returns:
+    spec: Filtered KfDef spec
+  """
+  for i, app in enumerate(spec["applications"]):
+    if app["name"] == "spartakus":
+      spec["applications"].pop(i)
+      break
+  return spec
+
+def get_config_spec(config_path, project, email, app_path):
+  """Generate KfDef spec.
+
+  Args:
+    config_path: Path to a YAML file containing a KFDef object.
+    Can be a local path or a URI like
+    https://raw.githubusercontent.com/kubeflow/manifests/master/kfdef/kfctl_gcp_iap.yaml
+    project: The GCP project to use.
+    email: a valid email of the GCP account
+    app_path: The path to the Kubeflow app.
+  Returns:
+    config_spec: Updated KfDef spec
+  """
+  # TODO(https://github.com/kubeflow/kubeflow/issues/2831): Once kfctl
+  # supports loading version from a URI we should use that so that we
+  # pull the configs from the repo we checked out.
+  config_spec = load_config(config_path)
+  config_spec["spec"]["project"] = project
+  config_spec["spec"]["email"] = email
+  config_spec["spec"] = filter_spartakus(config_spec["spec"])
+
+  # Set KfDef name to be unique
+  regex = re.compile('[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?')
+  kfdef_name = regex.findall(app_path)[-1]
+  config_spec["metadata"]["name"] = kfdef_name
+
+  repos = config_spec["spec"]["repos"]
+  if os.getenv("REPO_NAME") == "manifests":
+    # kfctl_go_test.py was triggered on presubmit from the kubeflow/manifests
+    # repository. In this case we want to use the specified PR of the
+    # kubeflow/manifests repository; so we need to change the repo specification
+    # in the KFDef spec.
+    # TODO(jlewi): We should also point to a specific commit when triggering
+    # postsubmits from the kubeflow/manifests repo
```

Comment:
Can we address this comment in this PR?
If not we can come back to it at a later point.

Reply:
I can include this, but need information on to change to what as I am still learning all the CI stuff that is happening here.

Reply:
Cool. We can open an issue and address it soon.